src/jit/hwintrinsicxarch.cpp

```diff
@@ -1030,6 +1030,23 @@ GenTree* Compiler::impSSE2Intrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_SSE2_StoreNonTemporal:
+        {
+            assert(sig->numArgs == 2);
+            assert(JITtype2varType(sig->retType) == TYP_VOID);
+            op2      = impPopStack().val;
+            op1      = impPopStack().val;
+            simdSize = ((op2->TypeGet() == TYP_LONG) || (op2->TypeGet() == TYP_ULONG)) ? 8 : 4;
+#ifdef _TARGET_X86_
+            if (simdSize == 8)
+            {
+                return impUnsupportedHWIntrinsic(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, method, sig, mustExpand);
```

Comment:
You do not need this code. Just give this intrinsic a flag `HW_Flag_SecondArgMaybe64Bit` is enough.